src/probability/extrealScript.sml

```diff
@@ -2890,14 +2892,28 @@ Proof
  >> MATCH_MP_TAC ln_neg_lt >> art []
 QED
 
+(* cf. transcTheory.LN_INV *)
+Theorem ln_inv :
+    !x. 0 < x ==> ln (inv x) = ~(ln x)
+Proof
+    rpt STRIP_TAC
+ >> ‘0 <= x’ by rw [le_lt]
+ >> ‘x <> NegInf’ by rw [pos_not_neginf]
+ >> Cases_on ‘x’ >> fs [extreal_ln_def, extreal_inv_def, extreal_ainv_def]
+ >> fs [extreal_of_num_def, extreal_lt_eq, extreal_le_eq]
+ >> ‘r <> 0’ by rw [REAL_LT_IMP_NE]
+ >> rw [extreal_inv_def, extreal_ln_def, extreal_ainv_def]
+ >> MATCH_MP_TAC LN_INV >> art []
+QED
+
 (***************************)
 (*      Exp and powr       *)
 (***************************)
 
 Theorem exp_pos :
     !x :extreal. 0 <= exp x
 Proof
-    GEN_TAC >> Cases_on `x`
+    Q.X_GEN_TAC ‘x’ >> Cases_on `x`
```

Comment:
If the name already occurs in the original goal (as here), using `Q.X_GEN_TAC` is unnecessarily paranoid.  (Also note that the easier to type `qx_gen_tac` also exists.)

Reply:
Yes, you are right (this is necessary). It's totally my (bad) personal style that 1) use upper-name tactics to distinguish between "inner" (logics) and "outer" languages (ML code), 2) use `X_GEN_TAC`  instead of `GEN_TAC` (and `Induct_on` instead of `Induct`, etc.) to make the proof a bit more readable.  But this is totally meaningless in very simple proofs like the present case.